### DISCUSSION TITLE
MES: some shift swaps not working with shift drop disabled

### DISCUSSION THREAD

**Discussion by emiljensen2** [runelite/runelite#15585] [Ideas]
If you have shift click drop disabled in osrs settings then some shift click swaps won't work on items in inventory, release on chins, and destroy on the water container in the zebac puzzle room is some I've noticed

**Comment by Enriath** [comment]
~~This is intended. Our shift-click stuff modifies how the vanilla shift-drop system works. If you don't want to drop an item while holding shift, you can shift right click the item and set something else to happen instead.~~

So this is all now wildly inconsistent

**Comment by emiljensen2** [comment]
So this bug is intended? Everything else works with shift click drops disabled, it's only these specific options on these items and possibly other items.
I can successfully set an item to shift click drop using mes and it'll work just fine.

some shift swaps simply don't work when set if you've set the game's shift click drop option to disabled.
>disable shift click drop in game settings
>shift right click chins and set them to release when holding shift
>doesn't work

Unfortunately your bandaid won't work for me since I'd have to swap the shift click on every item in the game then to avoid the game dropping my items when shift is not held.

**Comment by Enriath** [comment]
Moved to a discussion since there's now an inconsistency with how the shift stuff works.

**Comment by emiljensen2** [comment]
So testing a bit further it might be more impactful than i had noticed, for example on super combat I can shift swap to ""Use", and "Drink", but doesn't work for shift ""Empty, and "Drop"

**Comment by kamielvf** [comment]
I'm more surprised that some of the shift click options on items work when you have shift-drop disabled in the vanilla settings. When I added the functionality to modify shift clicks all it did was change a certain field on an item's composition and the vanilla client would handle the shift-swapping by itself. If you are able to modify certain swaps and they seem to work (when they shouldn't) that would be a vanilla "bug" (can't really call it a bug because the shift-drop feature is not intended to do anything other than drop).

Just to clear things up: 
1) you are required to enable shift-drop in vanilla settings for this feature to work
2) "release" and "destroy" are item options like drop, and as mentioned in point 1) you require shift-drop enabled in the vanilla settings for these to work

We handle the "use" shift-click on items ourselves, so if you disable the vanilla setting the only shift-click swap that would work is any item where you configure "use" as shift click.

Current behavior is as expected, tho I am not against changing it to where it works regardless of the vanilla setting

**Comment by emiljensen2** [comment]
Hmm that does make sense and sounds weird if it works even though it shouldn't.

I mean I wouldn't be against that either, tho I'd imagine very few people have their client setup like me
